### PR TITLE
Update PortablePdbReader.cs

### PR DESCRIPTION
**Pull request by jamescrosswell**
Fixes an issue in `PortablePdbReader.GetMetadataReader`. 

This method is supposed to return a value from the cache (if one exists) and otherwise try to create one from the assemblyPath. 

The previous code only ever did any work or added anything to the cache if the following condition was satisfied:
```csharp
 if (!_cache.TryGetValue(assemblyPath, out var provider) && provider is not null)
```

So if it couldn't find anything from the cache and also, if that thing it didn't find isn't null (which would never be possible - if it couldn't find anything, the out value will always be null).

### CODE REVIEW

**Comment by bruno-garcia** [comment]
CI is failing with:

>  error NU1903: Warning As Error: Package 'Microsoft.NETCore.App' 2.1.0 has a known high severity vulnerability,

Might need some fixing too (drop support for example?)

**Review by bruno-garcia — APPROVED** [review]

**Comment by bruno-garcia** [comment]
Worth opening a PR upstream on https://github.com/benaadams/Ben.Demystifier too

File: test/Ben.Demystifier.Test/MethodTests.cs [hunk 1]
```diff
@@ -77,11 +77,18 @@ public void DemistifiesMethodWithLambda()
             stackTrace = LineEndingsHelper.RemoveLineEndings(stackTrace);
             var trace = string.Join(string.Empty, stackTrace.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries));
 
+#if (NETFRAMEWORK && !DEBUG)
```
Comment (jamescrosswell) [inline test/Ben.Demystifier.Test/MethodTests.cs:80]: This appears unrelated to the changes in this PR btw... I suspect something obscure changed in the .NET Framework such that this is now required.
